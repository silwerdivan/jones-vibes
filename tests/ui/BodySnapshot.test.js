import fs from 'fs';
import path from 'path';

describe('Body Snapshot', () => {
  beforeAll(async () => {
    const html = fs.readFileSync(path.resolve(__dirname, '../../index.html'), 'utf8');
    document.body.innerHTML = html.toString();
    
    await import('../../js/app.js');

    document.dispatchEvent(new Event('DOMContentLoaded', { bubbles: true, cancelable: true }));
  });

  test('Body should match snapshot', () => {
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
