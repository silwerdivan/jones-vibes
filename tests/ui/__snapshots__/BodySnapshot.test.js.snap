// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Body Snapshot Test should match the snapshot of the initial body content 1`] = `
"
    <div class="game-container">
        <header class="game-title">
            <h1>JONES IN THE FAST LANE</h1>
        </header>
        <main class="main-grid">
            <div id="player-1" class="player-panel">
                <h2>Player 1</h2>
                <div class="stats">
                    <div class="stat-item"><span class="stat-label">Cash:</span> <span id="p1-cash" class="stat-value currency">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Savings:</span> <span id="p1-savings" class="stat-value">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Loan:</span> <span id="p1-loan" class="stat-value">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Happiness:</span> <span id="p1-happiness" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Education:</span> <span id="p1-education" class="stat-value">None</span></div>
                    <div class="stat-item"><span class="stat-label">Career:</span> <span id="p1-career" class="stat-value">Unemployed</span></div>
                    <div class="stat-item"><span class="stat-label">Car:</span> <span id="p1-car" class="stat-value">No</span></div>
                </div>
                <div class="time-remaining">Time Remaining: <span id="p1-time">0</span> hours</div>
            </div>

            <div id="player-2" class="player-panel">
                <h2>Player 2 (AI)</h2>
                <div class="stats">
                    <div class="stat-item"><span class="stat-label">Cash:</span> <span id="p2-cash" class="stat-value currency">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Savings:</span> <span id="p2-savings" class="stat-value">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Loan:</span> <span id="p2-loan" class="stat-value">$0</span></div>
                    <div class="stat-item"><span class="stat-label">Happiness:</span> <span id="p2-happiness" class="stat-value">0</span></div>
                    <div class="stat-item"><span class="stat-label">Education:</span> <span id="p2-education" class="stat-value">None</span></div>
                    <div class="stat-item"><span class="stat-label">Career:</span> <span id="p2-career" class="stat-value">Unemployed</span></div>
                    <div class="stat-item"><span class="stat-label">Car:</span> <span id="p2-car" class="stat-value">No</span></div>
                </div>
                <div class="time-remaining">Time Remaining: <span id="p2-time">0</span> hours</div>
            </div>

            <div id="location-info" class="info-panel">
                <h2>Current Location</h2>
                <p>Current Location: <span id="current-location">Start</span></p>
                <p>Turn: <span id="game-turn">1</span></p>
            </div>

            <div id="game-controls" class="actions-panel">
                <h2>Actions</h2>
                <div class="action-buttons">
                    <button id="btn-rest-end-turn">Rest / End Turn</button>
                    <button id="btn-travel" class="location-specific hidden">Travel</button>
                    <button id="btn-work-shift" class="location-specific hidden">Work Shift</button>
                    <button id="btn-take-course" class="location-specific hidden">Take Course</button>
                    <button id="btn-buy-item" class="location-specific hidden">Buy Item</button>
                    <button id="btn-deposit" class="location-specific hidden">Deposit</button>
                    <button id="btn-withdraw" class="location-specific hidden">Withdraw</button>
                    <button id="btn-take-loan" class="location-specific hidden">Take Loan</button>
                    <button id="btn-repay-loan" class="location-specific hidden">Repay Loan</button>
                    <button id="btn-buy-car" class="location-specific hidden">Buy Car</button>
                </div>
            </div>

            <div class="event-log" data-testid="game-log">
                <h3>Game Log</h3>
                <div class="log-content"></div>
            </div>
        </main>
    </div>

    <script type="module" src="js/app.js"></script>

"
`;
